### DISCUSSION TITLE
FLASK_APP is unavailable when FlaskGroup is not used

### DISCUSSION THREAD

**Discussion by TitaniumHocker** [pallets/flask#5156] [Q&A] [answered]
When cli group created with [flask.cli.AppGroup](https://github.com/pallets/flask/blob/main/src/flask/cli.py#L362) and returned to cli instead of default [flask.cli.FlaskGroup](https://github.com/pallets/flask/blob/main/src/flask/cli.py#L481) [flask.cli.ScriptInfo](https://github.com/pallets/flask/blob/main/src/flask/cli.py#L266) failing to locate application with `FLASK_APP` environment variable with this error message:
```
Error: Could not locate a Flask application. Use the 'flask --app' option, 'FLASK_APP' environment variable, or a 'wsgi.py' or 'app.py' file in the current directory.
```

Here is a simple example of executable file `main.py`:
```python
#!/usr/bin/env python
from os import environ

import click
from flask import Flask
from flask.cli import AppGroup

cli = AppGroup("main")


@cli.command()
def test():
    click.echo("Test command")


def create_app() -> Flask:
    app = Flask(__name__)
    app.cli.add_command(cli)
    return app


if __name__ == "__main__":
    environ.setdefault("FLASK_APP", "main:create_app")
    cli()
```

If run this file with flask<2.2.0:
```
$ ./main.py test
Test command
```
And with flask>=2.2.0:
```
$ ./main.py test
Usage: main.py test [OPTIONS]
Try 'main.py test --help' for help.

Error: Could not locate a Flask application. Use the 'flask --app' option, 'FLASK_APP' environment variable, or a 'wsgi.py' or 'app.py' file in the current directory.
```

I don't want to expand flask commands, so I don't use FlaskGroup, and use main.py script instead. This worked fine before 2.2.0.

I think this error is caused by [this](https://github.com/pallets/flask/commit/99fa3c36abc03cd5b3407df34dce74e879ea377a) commit. Before 2.2.0 ScriptInfo locates `app_import_path` automatically from environment variable `FLASK_APP`, after 2.2.0 it expects that this value will be provided by `--app` of option FlaskGroup that can get it from `FLASK_APP` environment variable.

Environment:

- Python version: 3.11
- Flask version: >=2.2.0

**Comment by davidism** [comment, marked as answer]
You still need to run commands with the `flask` command, that's why you register them with `app.cli.add_command`.

```
$ flask main test
```

If you really want to use only your group and not the `flask` command, then you need to create the `ScriptInfo` object like `FlaskGroup` would have done for you.

```python
if __name__ == "__main__":
    info = ScriptInfo(create_app=create_app)
    cli(obj=info)
```
